### PR TITLE
Fix graphite whitelist converter to handle array dimensions

### PR DESCRIPTION
**Pull request by kevinconaway**
Some service event metric dimensions are specified as arrays instead of keys.  The GC metrics from the `JvmMonitor` are an example of this:

```
2018-01-17T18:51:13,878 INFO [MonitorScheduler-0] com.metamx.emitter.core.LoggingEmitter - Event [{"feed":"metrics","timestamp":"2018-01-17T18:51:13.878Z","service":"druid/broker","host":"127.0.0.1:8082","version":"0.12.0-SNAPSHOT","metric":"jvm/gc/count","value":0,"gcGen":["young"],"gcName":["g1"]}]
2018-01-17T18:51:13,878 INFO [MonitorScheduler-0] com.metamx.emitter.core.LoggingEmitter - Event [{"feed":"metrics","timestamp":"2018-01-17T18:51:13.878Z","service":"druid/broker","host":"127.0.0.1:8082","version":"0.12.0-SNAPSHOT","metric":"jvm/gc/cpu","value":0,"gcGen":["young"],"gcName":["g1"]}]
```

If I attempt to use the _gcName_ or _gcGen_ attributes as part of the whitelist converter in the graphite emitter:

```javascript
{
  "jvm/gc": ["gcName"]
}
```

It will currently fail with

```
2018-01-17T19:06:44,993 ERROR [MonitorScheduler-0] com.metamx.common.concurrent.ScheduledExecutors - Uncaught exception.
java.lang.ClassCastException: java.util.Arrays$ArrayList cannot be cast to java.lang.String
	at io.druid.emitter.graphite.WhiteListBasedConverter.getOrderedDimValues(WhiteListBasedConverter.java:178) ~[?:?]
	at io.druid.emitter.graphite.WhiteListBasedConverter.druidEventToGraphite(WhiteListBasedConverter.java:213) ~[?:?]
	at io.druid.emitter.graphite.GraphiteEmitter.emit(GraphiteEmitter.java:105) ~[?:?]
	at com.metamx.emitter.service.ServiceEmitter.emit(ServiceEmitter.java:72) ~[java-util-1.3.2.jar:?]
	at com.metamx.emitter.service.ServiceEmitter.emit(ServiceEmitter.java:77) ~[java-util-1.3.2.jar:?]
	at com.metamx.metrics.JvmMonitor$GcGenerationCollector.emit(JvmMonitor.java:235) ~[java-util-1.3.2.jar:?]
	at com.metamx.metrics.JvmMonitor$GcGeneration.emit(JvmMonitor.java:200) ~[java-util-1.3.2.jar:?]
	at com.metamx.metrics.JvmMonitor$GcCounters.emit(JvmMonitor.java:157) ~[java-util-1.3.2.jar:?]
	at com.metamx.metrics.JvmMonitor.emitGcMetrics(JvmMonitor.java:128) ~[java-util-1.3.2.jar:?]
	at com.metamx.metrics.JvmMonitor.doMonitor(JvmMonitor.java:71) ~[java-util-1.3.2.jar:?]
	at com.metamx.metrics.AbstractMonitor.monitor(AbstractMonitor.java:41) ~[java-util-1.3.2.jar:?]
	at com.metamx.metrics.MonitorScheduler$1.call(MonitorScheduler.java:121) ~[java-util-1.3.2.jar:?]
	at com.metamx.metrics.MonitorScheduler$1.call(MonitorScheduler.java:116) ~[java-util-1.3.2.jar:?]
	at com.metamx.common.concurrent.ScheduledExecutors$4.run(ScheduledExecutors.java:173) [java-util-1.3.2.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_65]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [?:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [?:1.8.0_65]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_65]
```

This PR fixes the `WhiteListBasedConverter` to account for multi-valued dimensions

### CODE REVIEW

File: extensions-contrib/graphite-emitter/src/main/java/io/druid/emitter/graphite/WhiteListBasedConverter.java [hunk 1]
```diff
@@ -169,15 +170,26 @@ private String getPrefixKey(String key, SortedMap<String, ?> whiteList)
...
+    ImmutableList.Builder<String> outputList = new ImmutableList.Builder<>();
     Set<String> dimensions = whiteListDimsMapper.get(prefixKey);
     if (dimensions == null) {
       return Collections.emptyList();
     }
     for (String dimKey : dimensions) {
-      String dimValue = (String) event.getUserDims().get(dimKey);
-      if (dimValue != null) {
-        outputList.add(GraphiteEmitter.sanitize(dimValue));
+      Object rawValue = event.getUserDims().get(dimKey);
```
Comment (nishantmonu51) [inline extensions-contrib/graphite-emitter/src/main/java/io/druid/emitter/graphite/WhiteListBasedConverter.java]: Same fix also needs to be applied to WhiteListBasedDruidToTimelineEventConverter
@kevinconaway : can you fix that class too ?

Comment (kevinconaway) [reply extensions-contrib/graphite-emitter/src/main/java/io/druid/emitter/graphite/WhiteListBasedConverter.java]: Sure thing, done.

**Comment by b-slim** [comment]
👍  after tests

**Comment by kevinconaway** [comment]
What else needs to be done for this to be merged?

**Comment by nishantmonu51** [comment]
Thanks for the fix @kevinconaway

**Pull request merged**